packages/openapi-typescript/src/transform/schema-object.ts

```diff
@@ -447,7 +447,9 @@ function transformSchemaObjectCore(
         }
         let optional =
           schemaObject.required?.includes(k) ||
-          ("default" in v && options.ctx.defaultNonNullable)
+          ("default" in v &&
+            options.ctx.defaultNonNullable &&
+            !options.path?.includes("parameters")) // parameters can’t be required, even with defaults
```

Comment:
This seems like the dumbest solution to the problem, but I actually think it’s the correct one—parameters go through normal SchemaObject transformation, however, in this one scenario we apply different rules because of where they live (in any `parameters` object). I can’t think of a single scenario where this wouldn’t be true.

I also feel like the alternative—trying to override `ctx.defaultNonNullable` conditionally based on the parent—is a worse approach that could lead to other unintentional bugs.